### PR TITLE
chore(functions): fix flaky tests + update dir property

### LINKED ISSUE #5250: functions.ExampleIT: helloHttp_shouldRunWithFunctionsFramework failed
This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 6ba0025078a96bcc0646dd74fe4fdf36451cb4c1
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/ed65fc38-8416-4146-ba33-be937319a9ed), [Sponge](http://sponge2/ed65fc38-8416-4146-ba33-be937319a9ed)
status: failed
<details><summary>Test output</summary><br><pre>com.google.common.truth.AssertionErrorWithFacts: 
expected: Hello world!
but was an empty string
	at functions.hello.http@1.0.22/functions.ExampleIT.helloHttp_shouldRunWithFunctionsFramework(ExampleIT.java:89)
</pre></details>

### PR DESCRIPTION
**Pull request by ace-n**
This should fix #5250 and #5257

(I wasn't able to reproduce them locally, but this makes the tests themselves more fault-tolerant.)

### CODE REVIEW

**Review by lesv — CHANGES_REQUESTED** [review]

File: functions/helloworld/hello-gcs/src/test/java/functions/ExampleIT.java [hunk 1]
```diff
@@ -93,6 +93,14 @@ public void helloGcs_shouldRunWithFunctionsFramework() throws Throwable {
     RetryRegistry registry = RetryRegistry.of(RetryConfig.custom()
         .maxAttempts(8)
         .retryExceptions(HttpHostConnectException.class)
+        .retryOnResult(__ -> {
+          // Retry if the Functions Framework process has no stdout content
+          try {
+            return emulatorProcess.getErrorStream().available() == 0;
+          } catch (IOException e) {
+            return true;
+          }
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/helloworld/hello-gcs/src/test/java/functions/ExampleIT.java:102]: See `retryOnResultPredicate` on [this page](https://resilience4j.readme.io/docs/retry).

Happy to link to that in a comment, if that would help?

[the remark this answers is not in the document]
Comment (ace-n) [reply functions/helloworld/hello-gcs/src/test/java/functions/ExampleIT.java:102]: Yes - `maxAttempts(8)`

File: functions/helloworld/hello-pubsub/src/test/java/functions/ExampleIT.java [hunk 2]
```diff
@@ -91,6 +91,14 @@ public void helloPubSub_shouldRunWithFunctionsFramework() throws Throwable {
...
         .maxAttempts(8)
         .retryExceptions(HttpHostConnectException.class)
+        .retryOnResult(__ -> {
+          // Retry if the Functions Framework process has no stdout content
+          try {
+            return emulatorProcess.getErrorStream().available() == 0;
+          } catch (IOException e) {
+            return true;
+          }
+        })
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/helloworld/hello-pubsub/src/test/java/functions/ExampleIT.java:101]: As above - `maxAttempts(8)`

File: functions/helloworld/hello-gcs/src/test/java/functions/ExampleIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lesv) [reply functions/helloworld/hello-gcs/src/test/java/functions/ExampleIT.java:102]: Yes

[the remark this answers is not in the document]
Comment (ace-n) [reply functions/helloworld/hello-gcs/src/test/java/functions/ExampleIT.java:103]: Done, PTAL?

**Review by lesv — APPROVED** [review]

**Comment by lesv** [comment]
# Lint
```
#
# functions/helloworld/hello-gcs
#

[ERROR] src/test/java/functions/ExampleIT.java:[96,24] (naming) LambdaParameterName: Lambda parameter name '__' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'.
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (default-cli) on project functions-hello-gcs: You have 1 Checkstyle violation. -> [Help 1]


#
# functions/helloworld/hello-pubsub
#

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (default-cli) on project functions-hello-pub-sub: Execution default-cli of goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check failed: Plugin org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2 or one of its dependencies could not be resolved: Failed to collect dependencies at org.apache.maven.plugins:maven-checkstyle-plugin:jar:3.1.2 -> org.apache.maven:maven-core:jar:3.0 -> org.apache.maven:maven-aether-provider:jar:3.0 -> org.sonatype.aether:aether-api:jar:1.7: Failed to read artifact descriptor for org.sonatype.aether:aether-api:jar:1.7: Could not transfer artifact org.sonatype.aether:aether-parent:pom:1.7 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/sonatype/aether/aether-parent/1.7/aether-parent-1.7.pom.part (No such file or directory) -> [Help 1]
```

**Comment by ace-n** [comment]
@lesv thanks - see [this `checkstyle` change](https://github.com/GoogleCloudPlatform/java-repo-tools/pull/254).

**Review by lesv — APPROVED** [review]

**Pull request merged**